src/services/sync/strategies/WebUntis/WebUntisSchoolyearSyncer.js

```diff
@@ -0,0 +1,483 @@
+	}
+
+	/**
+	* @see {Syncer#steps}
+    *
+	* Perform either fetching from WebUntis and storing metadata into 'webuntisMetadata' or
+	* Creating actual schulcloud 'courses'.
+	*/
+	steps() {
+		this.logInfo(`Running WebUntis School Year Sync.\n`);
+
+		if (this.data.dryrun) {
+			return this.createMetaDataFromWebUntisSteps(this.data);
+		} else {
+			return this.createCoursesFromMetaDataAndWebUntisSteps(this.data);
+		}
+	}
+
+	/**
+	* 
+	*/
+	async createMetaDataFromWebUntisSteps(params) {
+		this.stats.success = false;
+		
+		const config = await this.obtainWebUntisConfig();
+
+		const session = await this.login(config);
+		const metaData = await this.fetchMetaData(session, params);
+		await this.logout(session);
+		await this.storeMetadata(metaData, params);
+
+		this.stats.success = true;
+	}
+
+	/**
+	* 
+	*/
+	async createCoursesFromMetaDataAndWebUntisSteps(params) {
+		this.stats.success = false;
+
+		const config = await this.obtainWebUntisConfig();
+
+		const metaData = await this.obtainMetadata(params);
+		const session = await this.login(config);
+
+		const data = await this.fetchData(session, metaData);
+
+		await this.logout(session);
+
+		await this.createCourses(data, params);
+
+		this.stats.success = true;
+	}
+
+	/**
+	* 
+	* @param {*} params 
+	*/
+	getImportConditionEvaluator(params) {
+		if (params.datatype === 'inclusive') {
+			return ((id) => params.courseMetadataIds.includes(id));
+		}
+		if (params.datatype === 'exclusive') {
+			return ((id) => !params.courseMetadataIds.includes(id));
+		}
+		throw new BadRequest('invalid datatype');
+	}
+
+	/**
+	* 
+	*/
+	async obtainWebUntisConfig() {
+		return {
+			url: '[URL]',
+			schoolname: '[schoolname]',
+			user: '[username]',
+			password: '[password]',
+		};
+	}
+
+	/**
+	* 
+	* @param {*} params 
+	*/
+	async obtainMetadata(params) {
+		const metaData = await this.app.service('webuntisMetadata').find({
+			query: { datasourceId: params.datasourceId },
+		});
+
+		return metaData;
+	}
+
+	/**
+	* 
+	* @param {*} config 
+	*/
+	async login(config) {
+		await super.login(config.url, config.schoolname, config.user, config.password);
+
+		return {
+			session: this.session,
+			rpc: this.rpc
+		};
+	}
+
+	/**
+	* 
+	* @param {*} session 
+	*/
+	async fetchMetaData(session) {
+		const intermediateData = {};
+		
+		intermediateData.currentSchoolYear = await this.getCurrentSchoolyear();
+
+		this.logger.info('Current school year ', intermediateData.currentSchoolYear);
+		
+		// To iterate over either concept
+		intermediateData.classes = await this.getClasses(intermediateData.currentSchoolYear.id);
+		intermediateData.teachers = await this.getTeachers();
+		intermediateData.rooms = await this.getRooms();
+		intermediateData.subjects = await this.getSubjects();
+
+		intermediateData.timeGrid = await this.getTimegrid();
+
+		this.logger.info('Classes ', intermediateData.classes);
+
+		// Initialize empty timetable for classes
+		for (let entry of intermediateData.classes) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['entry' is never reassigned. Use 'const' instead.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4061937)